run_arduino_gen.sh

```diff
@@ -0,0 +1,274 @@
+  if $(grep -q "stm32mp157c-ev" /proc/device-tree/compatible) ; then
+    BOARD="STM32MP157_EVAL"
+  elif $(grep -q "stm32mp157c-dk" /proc/device-tree/compatible) ; then
+    BOARD="STM32MP157_DK"
+  elif $(grep -q "stm32mp157a-dk" /proc/device-tree/compatible) ; then
+    BOARD="STM32MP157_DK"
+  elif $(grep -q "stm32mp157" /proc/device-tree/compatible) ; then
+    BOARD="STM32MP157_GENERIC"
+  else
+    echo "Board is not an STM32MP157 BOARD" > /dev/kmsg
+    exit 1
+  fi
+}
+
+
+firmware_load() {
+  if [ -z "$ELF_BINARY" ]; then
+    echo "No Arduino binary contained. Run generate command first."
+    exit 1
+  fi
+
+  if ( echo "$ELF_HASH $ELF_INSTALL_PATH" | sha256sum --status -c - 2>/dev/null ); then
+    # The same firmware already exists, skip this step
+    echo "The same firmware is already installed. Starting..."
+    return 0
+  fi
+
+  # Decode base64-encoded binary to a temp directory and check hash
+  tmp_elf_file="/tmp/$ELF_NAME"
+  if which uudecode >/dev/null 2>&1; then
+    echo -n "$ELF_BINARY" | uudecode -o /dev/stdout | gzip -d > "$tmp_elf_file"
+  else 
+    echo -n "$ELF_BINARY" | tail -n +2 | base64 -d - 2>/dev/null | gzip -d > "$tmp_elf_file"
+  fi
+  echo "$ELF_HASH $tmp_elf_file" | sha256sum --status -c -
+  
+  # Copy elf into /lib/firmware
+  mv $tmp_elf_file $ELF_INSTALL_PATH
+  echo "Arduino: Executable created: $ELF_INSTALL_PATH" > /dev/kmsg
+}
+
+
+firmware_start() {
+  # Change the name of the firmware
+  echo -n arduino.ino.elf > $REMOTEPROC_DIR/firmware
+
+  # Change path to found firmware
+  #echo -n /home/root >/sys/module/firmware_class/parameters/path
+
+  # Restart firmware
+  echo "Arduino: Starting $ELF_INSTALL_PATH" > /dev/kmsg
+  echo start > $REMOTEPROC_DIR/state 2>/dev/null || true
+}
+
+
+firmware_stop() {
+  # Stop the firmware
+  echo "Arduino: Stopping $ELF_INSTALL_PATH" > /dev/kmsg
+  echo stop > $REMOTEPROC_DIR/state 2>/dev/null || true
+}
+
+
+generate_packaged_script() {
+  elf_path="$1"
+  this_script=$(readlink -f "$0")
+  output_script="$2"
+  if [ "$this_script" = "$output_script" ]; then
+    echo "The output file name must be diffent from this script file"
+    exit 1
+  fi
+  
+  # Generate a copy of this script with a self-contained elf binary and its hash
+  # The elf binary is gzip'ed, making its size to 1/6, and then Base64-encoded
+  # using uuencode.
+  head -n $(grep -n "{%" "$this_script" | cut -d: -f1 | head -n 1) $this_script > $output_script 
+  echo "ELF_HASH='$(sha256sum $elf_path | cut -d' ' -f1)'" >> $output_script
+  echo -n "ELF_BINARY='" >> $output_script
+  if which uuencode >/dev/null 2>&1; then
+    gzip -c "$elf_path" | uuencode -m $ELF_NAME >> $output_script
+  else
+    echo "begin-base64 644 $ELF_NAME" >> $output_script
+    gzip -c "$elf_path" | base64 >> $output_script
+  fi
+  echo "'" >> $output_script
+  tail -n +$(grep -n "%}" "$this_script" | cut -d: -f1 | head -n 1) $this_script >> $output_script
+}
```

Comment:
This should be fine to set x mode
```suggestion
  chmod a+x $output_script
}
```

Reply:
I thought about it, but the `x` bit is likely to be removed while transferring to the target board, especially when the host is Windows.

Since one of the large target users of Arduino is Linux beginners, I slightly worry about unnecessary issues like "`./run_arduino.sh` doesn't work because permission rejected" just because the `x` bit is removed in some scenarios. That's why I insist on `sh run_arduino.sh` and never mention to use `./run_arduino.sh` in [the whole documentation](https://github.com/kbumsik/Arduino_Core_STM32/tree/stm32mp1/variants/STM32MP157_DK). On the other hand, experience users don't care much about it because they know when to set the x bit when they need.

Maybe I'm thinking too much about it.